driver-core/src/main/java/com/datastax/driver/core/Frame.java

```diff
@@ -328,4 +332,16 @@ protected void encode(ChannelHandlerContext ctx, Frame frame, List<Object> out)
             }
         }
     }
+
+    @ChannelHandler.Sharable
+    static class BetaVersionFlagAdder extends MessageToMessageEncoder<Frame> {
+        @Override
+        protected void encode(ChannelHandlerContext ctx, Frame frame, List<Object> out) throws Exception {
+            ProtocolVersion protocolVersion = frame.header.version;
+            // Check for the protocol version, if the downgrade has occurred during renegotiation
+            if (protocolVersion.toInt() >= ProtocolVersion.V5.toInt())
+                frame.header.flags.add(Header.Flag.USE_BETA);
```

Comment:
Possibly related to this: how does auto-negotiation behave if
* driver is an older version that supports a beta v5
* Cassandra is a newer version that has the complete v5
* you set allowBetas = false

Could negotiation somehow succeed and result in v5/no beta? That would be bad because the driver could fail on some of the not-yet-implemented-on-the-client v5 features.

I feel like the driver should have the notion of whether its latest supported version is fully implemented or not (and not even allow non-beta attempts if it is not).